sources/packages/backend/apps/workers/src/services/disbursement-schedule/disbursement-schedule-service.ts

```diff
@@ -86,24 +256,516 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
         ASSESSMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE,
       );
     }
+    return assessment;
+  }
 
-    for (const disbursement of disbursements) {
-      const newDisbursement = new DisbursementSchedule();
-      newDisbursement.disbursementDate = disbursement.disbursementDate;
-      newDisbursement.negotiatedExpiryDate = disbursement.negotiatedExpiryDate;
-      newDisbursement.disbursementValues = disbursement.disbursements.map(
-        (disbursementValue) => {
-          const newValue = new DisbursementValue();
-          newValue.valueType = disbursementValue.valueType;
-          newValue.valueCode = disbursementValue.valueCode;
-          newValue.valueAmount = disbursementValue.valueAmount.toString();
-          return newValue;
+  /**
+   * Rollback overawards from the current and future applications.
+   * Past pending assessment will not be impacted.
+   * @param studentId student id.
+   * @param offeringStartDate offering start date used as a reference
+   * to find all applications in the future.
+   * @param entityManager used to execute the commands in the same transaction.
+   */
+  async rollbackOverawards(
+    studentId: number,
+    offeringStartDate: string,
+    entityManager: EntityManager,
+  ): Promise<void> {
+    // Submitted applications are not allowed to have overlaps what makes possible
+    // to use the offering start date to order them.
```

Comment:
`order them`.mm I didn't understand

Reply:
When rolling back we need to revert the current application being reassessed and the future ones. To know which ones we need to rollback, we get all applications ordered by the offering start date.
The point here is mainly that while in development if using the BYPASS_APPLICATION_SUBMIT_VALIDATIONS can mess up this process because the order can not happen properly.

Reply:
oh..okay....here the current application is realtive, right? it's the application that is being edited/canceled right? if so sure, not sure if it is a real scenario. suppose the student has 2 applications, both offering start is for the future, For example, `application 1` start date is April 2023, and  `application 1` start date is April 2024. And he/she canceled/updated the 2nd application. at that point will we reassess the first one?

Reply:
Nope, we will not reassess the application before the one being reassessed.
Considering the below:
- **Application A**: April 2023
- **Application B**: April 2024
- **Application C**: July 2024

When **Application A** changes, **Application A, B, and C** are reassessed.
When **Application B** changes, **Application B, and C** are reassessed.
When **Application B** changes, **Application C** is reassessed.